File: android/core/ui/src/main/kotlin/com/dualmusic/core/ui/components/DMLogo.kt
package com.dualmusic.core.ui.components

import androidx.compose.foundation.Image
import androidx.compose.foundation.layout.height
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.dualmusic.core.ui.R

/**
 * Logo Dual Music.
 *
 * Affiche EXACTEMENT le même logo que le site web (asset `logo-tr.png` réutilisé), pour
 * garantir la reconnaissance de marque entre plateformes. Le ratio d'origine est
 * préservé ; on contrôle uniquement la hauteur.
 *
 * @param modifier modificateur Compose (positionnement/alignement par l'appelant).
 * @param height hauteur cible du logo (la largeur s'ajuste au ratio). Défaut : 64 dp.
 */
@Composable
fun DMLogo(
    modifier: Modifier = Modifier,
    height: Dp = 64.dp,
) {
    Image(
        painter = painterResource(id = R.drawable.dm_logo),
        contentDescription = "Dual Music",
        contentScale = ContentScale.Fit,
        modifier = modifier.height(height),
    )
}

File: ios/Packages/CoreUI/Sources/CoreUI/Components/DMLogo.swift
import SwiftUI

/// Logo Dual Music (iOS).
///
/// Affiche **exactement le même logo que le site web** (asset `logo-tr.png` réutilisé,
/// embarqué dans `CoreUI` via `Bundle.module`), pour la parité de marque entre
/// plateformes. Le ratio d'origine est préservé ; l'appelant contrôle la hauteur.
public struct DMLogo: View {
    /// Hauteur cible du logo (la largeur s'ajuste au ratio).
    private let height: CGFloat

    /// - Parameter height: hauteur en points (défaut 64).
    public init(height: CGFloat = 64) {
        self.height = height
    }

    public var body: some View {
        // Chargé depuis le bundle du package (résilient si l'asset venait à manquer).
        (dmLogoImage ?? Image(systemName: "music.note"))
            .resizable()
            .scaledToFit()
            .frame(height: height)
            .accessibilityLabel("Dual Music")
    }

    /// Charge l'image du logo depuis les ressources du package.
    private var dmLogoImage: Image? {
        #if canImport(UIKit)
        if let ui = UIImage(named: "dm_logo", in: .module, with: nil) {
            return Image(uiImage: ui)
        }
        #endif
        return nil
    }
}
